core/object.cpp

```diff
@@ -1581,7 +1581,7 @@ void Object::_bind_methods() {
 
 	ClassDB::bind_method(D_METHOD("is_queued_for_deletion"), &Object::is_queued_for_deletion);
 
-	ClassDB::add_virtual_method("Object", MethodInfo("free"), false);
+	ClassDB::add_virtual_method("Object", MethodInfo("_free"), false);
```

Comment:
Regardless of what the consensus about this changes is, this method in particular is not a normal virtual method. Notice the second parameter is `false`. This tells ClassDB to add it to the virtual method list but to not add the flag `METHOD_FLAG_VIRTUAL`, hence the method is not virtual.

AFAIK, this hack is needed because `free` must be called before locking the object for normal method calls. See:

https://github.com/godotengine/godot/blob/c16d00591be623e0b1d6377e6dda7007c0e0d6c4/core/object.cpp#L637-L676